### PR TITLE
Prevent users from selecting same asset in/out for a swap

### LINKED ISSUE #533: Should not be able to select same asset in/out
User's should not be able to select the same asset for the in/out box. Will involve filtering and a smarter way to initialize the component.

Screenshot 2024-02-15 at 3 00 53 PM

This could be part of `assertValidSwaps` in the custody service's `authorize` RPC method. See https://github.com/penumbra-zone/web/pull/821#discussion_r1537044452

### PR DESCRIPTION
**Pull request by jessepinho**
Per #533, we want to prevent users from attempting a swap in which both the input and output asset types are the same.

This PR does that at two levels: in the extension's `authorize` RPC method, and in the UI. This ensures that the extension will prevent such swaps even if executed by a frontend other than minifront.

First, I disabled my changes to the UI so that I could still attempt a swap with the same asset in and out. Here's the error the extension gives:
image

Of course, no one on minfront will see this error, since we're now preventing the user from selecting the same asset in/out:

https://github.com/penumbra-zone/web/assets/1121544/1b773907-ba54-4c55-8c18-ac4e840c050f

Also in this PR, I added sorting to the asset in/out selectors, since the asset in selector was picking its default asset based on which balance happened to load first, and the asset out selector was using the sorting in the registry.

Closes #533

### CODE REVIEW

File: apps/minifront/src/components/swap/swap-loader.tsx [hunk 1]
```diff
@@ -29,11 +32,18 @@ const getAndSetDefaultAssetBalances = async () => {
   const assetBalances = await getBalances();
 
   // filter assets that are not available for swap
-  const filteredAssetBalances = assetBalances.filter(b =>
-    [assetPatterns.lpNft, assetPatterns.proposalNft, assetPatterns.votingReceipt].every(
-      pattern => !pattern.matches(getDisplayDenomFromView(b.balanceView)),
-    ),
-  );
+  const filteredAssetBalances = assetBalances
```
[the remark this answers is not in the document]
Comment (erwanor) [reply apps/minifront/src/components/swap/swap-loader.tsx:35]: Yeah this makes sense it’s common UX for decentralized exchanges

[the remark this answers is not in the document]
Comment (jessepinho) [reply apps/minifront/src/components/swap/swap-loader.tsx:46]: Fixed!

File: packages/router/src/grpc/custody/authorize/index.ts [hunk 2]
```diff
@@ -22,6 +23,7 @@ export const authorize: Impl['authorize'] = async (req, ctx) => {
   assertSwapClaimAddressesBelongToCurrentUser(req.plan, address =>
     isControlledAddress(fullViewingKey, address),
   );
+  assertSwapAssetsAreNotTheSame(req.plan);
```
[the remark this answers is not in the document]
Comment (Valentine1898) [reply packages/router/src/grpc/custody/authorize/index.ts:26]: Uh, yeah, sure

[the remark this answers is not in the document]
Comment (jessepinho) [reply packages/router/src/grpc/custody/authorize/index.ts:26]: @Valentine1898 I actually think we shouldn't move the assertions to the planning stage for security reasons. I put this code in a separate PR [here](https://github.com/penumbra-zone/web/pull/869); please leave a comment re: whether my thinking is valid or not.

**Pull request merged**